lib/ruby_lsp/requests/support/annotation.rb

```diff
@@ -0,0 +1,76 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    module Support
+      class Annotation
+        extend T::Sig
+        sig do
+          params(
+            arity: T.any(Integer, T::Range[Integer]),
+            block: Block,
+            receiver: T::Boolean,
+          ).void
+        end
+        def initialize(arity:, block:, receiver: false)
+          @arity = arity
+          @block = block
+          @receiver = receiver
+        end
+        sig { returns(T.any(Integer, T::Range[Integer])) }
```

Comment:
(nit) Normally I'd expect to see any `attr_*`s before the methods in class.

Reply:
It may not be universal, but it seems to be the convention we use in this repo.

It also what the Ruby Style Guide [suggests](https://github.com/rubocop/ruby-style-guide#consistent-classes), although the Shopify one doesn't mention it.